components/script/body.rs

```diff
@@ -891,6 +895,23 @@ fn run_form_data_algorithm(
     Err(Error::Type("Inappropriate MIME-type for Body".to_string()))
 }
 
+#[allow(unsafe_code)]
+fn run_bytes_data_algorithm(cx: JSContext, bytes: Vec<u8>) -> Fallible<FetchedData> {
+    rooted!(in(*cx) let mut array_buffer_ptr = ptr::null_mut::<JSObject>());
+    let arraybuffer = unsafe {
+        Uint8Array::create(
+            *cx,
+            CreateWith::Slice(&bytes),
+            array_buffer_ptr.handle_mut(),
+        )
+    };
```

Comment:
I think you can use [`create_buffer_source`](https://doc.servo.org/script/dom/bindings/buffer_source/fn.create_buffer_source.html) to avoid unsafe code here.

For reference, subtlecrypto does something similar:
https://github.com/servo/servo/blob/3122de08f3571a4026e825dd3ef6412cb37abfae/components/script/dom/subtlecrypto.rs#L320-L322

Reply:
Thank you! Updated, as well as replaced the unsafe code in `run_array_buffer_data_algorithm` below this.

[Updated Try](https://github.com/shanehandley/servo/actions/runs/13095826074).